presto-spi/src/main/java/com/facebook/presto/spi/function/TypeVariableConstraint.java

```diff
@@ -91,12 +92,13 @@ public boolean canBind(Type type)
         if (orderableRequired && !type.isOrderable()) {
             return false;
         }
-        if (!typeBound.isInstance(type)) {
+        if (variadicBound != null && !UNKNOWN.equals(type) && !type.getTypeSignature().getBase().equals(variadicBound)) {
```

Comment:
What's the reason of this change?

Reply:
Because it was a hack due to UnknwonType not being available typeConstaint. So the main reason for moving UnknwonType is to be able to use it in this classs.

Reply:
Change the commit message to reflect this then.

Reply:
Done